mlrun/runtimes/package/context_handler.py

```diff
@@ -0,0 +1,709 @@
+        # `matplotlib` and `seaborn`:
+        try:
+            import matplotlib.pyplot as plt
+
+            from mlrun.artifacts import PlotArtifact
+
+            # Get the figure:
+            figure = None
+            if isinstance(obj, plt.Figure):
+                figure = obj
+            elif isinstance(obj, plt.Axes):
+                if hasattr(obj, "get_figure"):
+                    figure = obj.get_figure()
+                elif hasattr(obj, "figure"):
+                    figure = obj.figure
+                elif hasattr(obj, "fig"):
+                    figure = obj.fig
+
+            # Create the artifact:
+            if figure is not None:
+                artifact = PlotArtifact(key=key, body=figure)
+        except ModuleNotFoundError:
+            pass
+
+        # `plotly`:
+        if artifact is None:
+            try:
+                import plotly
+
+                from mlrun.artifacts import PlotlyArtifact
+
+                if isinstance(obj, plotly.graph_objs.Figure):
+                    artifact = PlotlyArtifact(key=key, figure=obj)
+            except ModuleNotFoundError:
+                pass
+
+        # `bokeh`:
+        if artifact is None:
+            try:
+                import bokeh.plotting as bokeh_plt
+
+                from mlrun.artifacts import BokehArtifact
+
+                if isinstance(obj, bokeh_plt.Figure):
+                    artifact = BokehArtifact(key=key, figure=obj)
+            except ModuleNotFoundError:
+                pass
+            except ImportError:
+                logger.warn(
+                    "Bokeh installation is ignored. If needed, "
+                    "make sure you have the required version with `pip install mlrun[bokeh]`"
+                )
+
+        # Log the artifact:
+        if artifact is None:
+            raise MLRunInvalidArgumentError(
+                f"The given plot is of type `{type(obj)}`. We currently support logging plots produced by one of "
+                f"the following modules: `matplotlib`, `seaborn`, `plotly` and `bokeh`. You may try to save the "
+                f"plot to file and log it as a file instead."
+            )
+        ctx.log_artifact(**logging_kwargs, item=artifact)
+
+    @staticmethod
+    def log_result(
+        ctx: MLClientCtx,
+        obj: Union[int, float, str, list, tuple, dict, np.ndarray],
+        key: str,
+        logging_kwargs: dict,
+    ):
+        """
+        Log an object as a result. The objects value will be cast to a serializable version of itself. Supporting:
+        int, float, str, list, tuple, dict, numpy.ndarray
+
+        :param ctx:            The MLRun context to log with.
+        :param obj:            The value to log.
+        :param key:            The key of the artifact.
+        :param logging_kwargs: Additional keyword arguments to pass to the `context.log_result` method.
+        """
+        ctx.log_result(**logging_kwargs, key=key, value=obj)
+
+
+class ContextHandler:
```

Comment:
if private than I would rename it to 
```suggestion
class _ContextHandler:
```

Reply:
well, I use it outside of the file it is declared at, so it is not private in that regard. It is private in that users should not import it, that I agree. Once I'll write the package module and the decorator will be inside of it, I'll mark it with `_` for sure, no worries.